Reduce use of downcast<>() in WebCore/html

cdumez opened the pull request:
#### 711120e7edec012527620d07bf63d85713a180fd
<pre>
Reduce use of downcast&lt;&gt;() in WebCore/html
<a href="https://bugs.webkit.org/show_bug.cgi?id=270761">https://bugs.webkit.org/show_bug.cgi?id=270761</a>

Reviewed by Darin Adler.

* Source/WebCore/html/HTMLCanvasElement.cpp:
(WebCore::HTMLCanvasElement::createContextWebGL):
(WebCore::HTMLCanvasElement::getContextWebGL):
(WebCore::HTMLCanvasElement::createContextBitmapRenderer):
(WebCore::HTMLCanvasElement::reset):
* Source/WebCore/html/HTMLMediaElement.cpp:
(WebCore::HTMLMediaElement::setVideoFullscreenStandby): Deleted.
* Source/WebCore/html/HTMLMediaElement.h:
(WebCore::HTMLMediaElement::videoFullscreenStandby const):
(WebCore::HTMLMediaElement::setVideoFullscreenStandbyInternal):
* Source/WebCore/html/HTMLVideoElement.cpp:
(WebCore::HTMLVideoElement::setVideoFullscreenStandby):
* Source/WebCore/html/HTMLVideoElement.h:
* Source/WebCore/html/canvas/WebGLUtilities.h:
(WebCore::ScopedWebGLRestoreFramebuffer::~ScopedWebGLRestoreFramebuffer):
* Source/WebCore/html/track/TrackEvent.cpp:
(WebCore::convertToTrackEventTrack):

Canonical link: <a href="https://commits.webkit.org/275899@main">https://commits.webkit.org/275899@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/65445d5e1acd432c4aa9ba8c918ade003620251c

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/43190 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/22214 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/45590 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/45821 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/39313 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/45495 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/25964 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/19637 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/35712 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/43763 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/19263 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/37219 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/16708 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/16849 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/38279 "Passed tests") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/1244 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/39388 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/38600 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/47367 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/18104 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/14898 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/42506 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/19641 "Built successfully") | | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/41166 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/19819 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/5861 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/19273 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

darinadler commented on Source/WebCore/html/HTMLCanvasElement.cpp:
```diff
@@ -433,18 +433,20 @@ WebGLRenderingContextBase* HTMLCanvasElement::createContextWebGL(WebGLVersion ty
-    m_context = WebGLRenderingContextBase::create(*this, attrs, type);
-    if (m_context) {
+    auto context = WebGLRenderingContextBase::create(*this, attrs, type);
+    WeakPtr weakContext = context.get();
```
Can this use `RefPtr` instead?

cdumez replied:
WebGLRenderingContextBase is not refcounted. create() returns a unique_ptr.

darinadler commented on Source/WebCore/html/HTMLCanvasElement.cpp:
```diff
@@ -477,15 +480,17 @@ ImageBitmapRenderingContext* HTMLCanvasElement::createContextBitmapRenderer(cons
-    m_context = ImageBitmapRenderingContext::create(*this, WTFMove(settings));
-    downcast<ImageBitmapRenderingContext>(m_context.get())->transferFromImageBitmap(nullptr);
+    auto context = ImageBitmapRenderingContext::create(*this, WTFMove(settings));
+    WeakPtr weakContext = *context;
```
Can this use `RefPtr` instead?

cdumez replied:
Ditto.

cdumez committed:
Reduce use of downcast<>() in WebCore/html
https://bugs.webkit.org/show_bug.cgi?id=270761

Reviewed by Darin Adler.

* Source/WebCore/html/HTMLCanvasElement.cpp:
(WebCore::HTMLCanvasElement::createContextWebGL):
(WebCore::HTMLCanvasElement::getContextWebGL):
(WebCore::HTMLCanvasElement::createContextBitmapRenderer):
(WebCore::HTMLCanvasElement::reset):
* Source/WebCore/html/HTMLMediaElement.cpp:
(WebCore::HTMLMediaElement::setVideoFullscreenStandby): Deleted.
* Source/WebCore/html/HTMLMediaElement.h:
(WebCore::HTMLMediaElement::videoFullscreenStandby const):
(WebCore::HTMLMediaElement::setVideoFullscreenStandbyInternal):
* Source/WebCore/html/HTMLVideoElement.cpp:
(WebCore::HTMLVideoElement::setVideoFullscreenStandby):
* Source/WebCore/html/HTMLVideoElement.h:
* Source/WebCore/html/canvas/WebGLUtilities.h:
(WebCore::ScopedWebGLRestoreFramebuffer::~ScopedWebGLRestoreFramebuffer):
* Source/WebCore/html/track/TrackEvent.cpp:
(WebCore::convertToTrackEventTrack):

Canonical link: https://commits.webkit.org/275899@main
